### PR TITLE
[MINOR] Fix Typos 

### PR DESCRIPTION
**Pull request by zhengruifeng**
## What changes were proposed in this pull request?

1,Rename matrix args in BreezeUtil to upper to match the doc
2,Fix several typos in ML and SQL
## How was this patch tested?

manual tests

### CODE REVIEW

**Comment by shaneknapp** [comment]
i will retrigger this build once maintenance is over.

File: mllib/src/main/scala/org/apache/spark/ml/ann/BreezeUtil.scala [hunk 1]
```diff
@@ -55,7 +55,7 @@ private[ann] object BreezeUtil {
    * @param y y
    */
   def dgemv(alpha: Double, a: BDM[Double], x: BDV[Double], beta: Double, y: BDV[Double]): Unit = {
-    require(a.cols == x.length, "A & b Dimension mismatch!")
+    require(a.cols == x.length, "A & x Dimension mismatch!")
```
Comment (srowen) [inline mllib/src/main/scala/org/apache/spark/ml/ann/BreezeUtil.scala]: Y'know, while we're at it, the matrix args in this whole class don't match the docs. The arg here and similarly elsewhere can be called "A" not "a".

Also this should check that `A.rows == y.length` for consistency.

File: mllib/src/main/scala/org/apache/spark/ml/ann/Layer.scala [hunk 2]
```diff
@@ -64,8 +64,9 @@ private[ann] trait Layer extends Serializable {
    * @return the layer model
    */
   def createModel(initialWeights: BDV[Double]): LayerModel
+
   /**
-   * Returns the instance of the layer with random generated weights
+   * Returns the instance of the layer with random generated weights.
```
Comment (srowen) [inline mllib/src/main/scala/org/apache/spark/ml/ann/Layer.scala]: This could just become the `@return` text which is otherwise unuseful

**Comment by shaneknapp** [comment]
jenkins, test this please

**Comment by shaneknapp** [comment]
jenkins, test this please

File: mllib/src/main/scala/org/apache/spark/ml/ann/BreezeUtil.scala (hunk 1 above)
Comment (zhengruifeng) [inline mllib/src/main/scala/org/apache/spark/ml/ann/BreezeUtil.scala]: Right. I will rename the matrix args to upper and add this missing checking.

File: mllib/src/main/scala/org/apache/spark/ml/ann/Layer.scala (hunk 2 above)
Comment (zhengruifeng) [inline mllib/src/main/scala/org/apache/spark/ml/ann/Layer.scala]: This method already have a `@return` text.

Comment (srowen) [inline mllib/src/main/scala/org/apache/spark/ml/ann/Layer.scala]: Yes, I see, but it just says "returns the layer model", which is what the text above already says in more detail. I didn't see a point in repeating it

**Comment by rxin** [comment]
@srowen we should backport the doc fixes into branch-2.0.

**Pull request closed without merging**